aesara/link/jax/dispatch/random.py

```diff
@@ -277,3 +277,49 @@ def sample_fn(rng, size, dtype, *parameters):
         return (rng, sample)
 
     return sample_fn
+
+
+@jax_sample_fn.register(aer.MultinomialRV)
+def jax_sample_fn_multinomial(op):
+    """JAX implementation of `MultinomialRV`."""
+
+    def _scatter_add_one(operand, indices, updates):
+        outcome_cnts = jax.lax.scatter_add(
+            operand, indices, updates, 
+            jax.lax.ScatterDimensionNumbers(
+                update_window_dims=(),
+                inserted_window_dims=(0,),
+                scatter_dims_to_operand_dims=(0,)
+                )
+            )
+
+        return outcome_cnts
+
+    def _categorical(key, p, shape):
+        shape = shape or p.shape[:-1]
+        s = jax.numpy.cumsum(p, axis=-1)
+        r = jax.random.uniform(key, shape=shape + (1,))
+        
+        return jax.numpy.sum(s < r, axis=-1)
```

Comment:
I am surprised because JAX does have an implementation for the categorical distribution [here](https://jax.readthedocs.io/en/latest/_autosummary/jax.random.categorical.html#jax.random.categorical) that uses their implementation for the Gumbel distribution. Is this justified in the codebase, or is it just because it was implemented at a time where `jax.random.categorical` was not available (which you should be able to determine with `git blame`)?